### PR TITLE
Delay platform view removal to submitFrame.

### PR DESCRIPTION
**Pull request by cyanglaz**
Remove platform views inside OnDispose might cause issue if it happens before SubmitFrame and the active_composition_order has not been updated.
We will be left at a situation where the active_composition_order still contains the old view, but the view is actually removed from the UIView hierarchy. 

We now cache the views need to be removed in OnDispose and actually remove them in SubmitFrame. At the same time, we remove stop detaching the subviews if they were disposed. 

As a side effect, this update will also fix https://github.com/flutter/flutter/issues/30220

### CODE REVIEW

**Comment by domesticmouse** [comment]
Any ideas on when this would land on dev branch?

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm [hunk 1]
```diff
@@ -257,6 +253,25 @@
 }
 
 void FlutterPlatformViewsController::DetachUnusedLayers() {
+  // Clean up disposed views.
+  if (!views_waiting_to_dispose_.empty()) {
+    for (std::vector<int64_t>::iterator it = active_composition_order_.begin(); it != active_composition_order_.end();) {
```
Comment (amirh) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: we should dispose the view even if it's not in `active_composition_order_`

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h [hunk 2]
```diff
@@ -91,6 +91,8 @@ class FlutterPlatformViewsController {
   GrContext* overlays_gr_context_;
   SkISize frame_size_;
 
+  std::unordered_set<int64_t> views_waiting_to_dispose_;
```
Comment (amirh) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h]: nit: views_to_dispose_
Please also add a comment explaining that method channel dispose calls adds the view to this set to be disposed on the next frame.

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm (hunk 1 above)
Comment (cyanglaz) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: Make sense, I will fix it on Monday.

Comment (cyanglaz) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: Done!

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h (hunk 2 above)
Comment (cyanglaz) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h]: Done

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm [hunk 3]
```diff
@@ -257,6 +253,23 @@
 }
 
 void FlutterPlatformViewsController::DetachUnusedLayers() {
+  // Clean up disposed views.
```
Comment (amirh) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: It's possible that the composition order didn't change, in which case we will return on line 229 and won't dispose the views that should be disposed.

I'd suggest splitting the part that disposes views out of this method, and call `DisposeViews` earlier in `SubmitFrame`, as this method currently is doing more than what its name suggests.

Comment (cyanglaz) [reply shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: Done

**Comment by qimingweng** [comment]
How can we tell when merges like this land on dev/beta or a hotfix?

**Pull request merged**